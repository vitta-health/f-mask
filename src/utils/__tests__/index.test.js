import { queryInputElementInside } from '../index';

describe('queryInputElementInside', () => {
  it('is a function', () => {
    expect(queryInputElementInside).toEqual(expect.any(Function));
  });

  it('returns the same HTMLInputElement when HTMLInputElement is passed', () => {
    const input = document.createElement('input');
    input.type = 'text';
    expect(queryInputElementInside(input)).toBe(input);
  });

  it('returns inner HTMLInputElement when HTMLElement with input is passed', () => {
    const root = document.createElement('div');
    const input = document.createElement('input');
    root.appendChild(input);
    expect(queryInputElementInside(root)).toBe(input);
  });

  it('returns original HTMLElement when HTMLElement without input is passed', () => {
    const root = document.createElement('div');
    const p = document.createElement('p');
    const span = document.createElement('span');
    p.appendChild(span);
    root.appendChild(p);
    expect(queryInputElementInside(root)).toBe(root);
  });

  it('throws an error when not HTMLElement is passed', () => {
    expect(() => queryInputElementInside('')).toThrowErrorMatchingSnapshot();
  });
});
