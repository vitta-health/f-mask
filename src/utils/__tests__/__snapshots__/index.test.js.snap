// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryInputElementInside throws an error when not HTMLElement is passed 1`] = `"el.querySelector is not a function"`;
